llvm/include/llvm/IR/DIBuilder.h

```diff
@@ -458,15 +458,18 @@ namespace llvm {
     /// \param Elements     Struct elements.
     /// \param RunTimeLang  Optional parameter, Objective-C runtime version.
     /// \param UniqueIdentifier A unique identifier for the struct.
+    /// \param SpecificationOf The type that this type completes (is a
+    /// specification of). For example, this could be a templated type whose
+    /// template parameters have been substituted in.
     /// \param NumExtraInhabitants The number of extra inhabitants of the type.
     /// An extra inhabitant is a bit pattern that does not represent a valid
     /// value for objects of a given type.
     DICompositeType *createStructType(
         DIScope *Scope, StringRef Name, DIFile *File, unsigned LineNumber,
         uint64_t SizeInBits, uint32_t AlignInBits, DINode::DIFlags Flags,
         DIType *DerivedFrom, DINodeArray Elements, unsigned RunTimeLang = 0,
-        DIType *VTableHolder = nullptr, StringRef UniqueIdentifier = "", 
-        uint32_t NumExtraInhabitants = 0);
+        DIType *VTableHolder = nullptr, StringRef UniqueIdentifier = "",
+        DIType *SpecificationOf = nullptr, uint32_t NumExtraInhabitants = 0);
```

Comment:
Clang already emits some composite types with a DW_AT_specification. What does the input IR for this look like and specificially, could this field here interfer with that mechanism?

Reply:
IIUC, this is only done for DISubprogram, and uses the declaration: field for that. Sounds like there is no potential for misunderstanding.